### PR TITLE
[ETH-FLOW V2] - Base Feature Branch

### PR DESCRIPTION
**Pull request by W3stside**
# Summary
Eth-Flow V2 feature branch

### This CANNOT be merged into develop UNTIL the backend/entire ETH-FLOW is finished. For now this is the base branch.

Includes:
- [x] Slippage label
- [x] Slippage updater
- [x] Banner component

Incoming
- [ ] contracts and hooks
- [ ] order cancellation and modals
- [ ] order state tracking
- [ ] valid_to updater

### CODE REVIEW

File: src/cow-react/modules/swap/containers/RowSlippage/index.tsx [hunk 1]
```diff
@@ -5,22 +5,34 @@ import { PERCENTAGE_PRECISION } from 'constants/index'
...
   showSettingOnClick?: boolean
 }
 
-export function RowSlippage({ allowedSlippage, showSettingOnClick = true }: RowSlippageProps) {
+export function RowSlippage({ trade, allowedSlippage, showSettingOnClick = true }: RowSlippageProps) {
   const toggleSettings = useToggleSettingsMenu()
+
+  // if is wrap/unwrap operation return null
+  const wrapType = useWrapType()
+  // should we show the warning?
```
Comment (josojo) [inline src/cow-react/modules/swap/containers/RowSlippage/index.tsx:22]: Which warning are you referring to?

File: src/cow-react/modules/swap/pure/Row/RowSlippageContent/index.tsx [hunk 2]
```diff
@@ -43,20 +56,35 @@ export function RowSlippageContent(props: RowSlippageContentProps) {
...
-                The slippage you pick here enables a resubmission of your order in case of unfavourable price movements.
-              </p>
-              <p>{INPUT_OUTPUT_EXPLANATION}</p>
-            </Trans>
+            showEthFlowSlippageWarning ? (
+              <Trans>
+                <p>You are currently swapping {symbols?.[0] || 'a native token'}.</p>
+                <p>
+                  Slippage tolerance is defaulted to 2% to ensure a high likelihood of order matching, even in volatile
+                  market situations.
```
Comment (josojo) [inline src/cow-react/modules/swap/pure/Row/RowSlippageContent/index.tsx:64]: Maybe we add:
```
Slippage tolerance is defaulted to 2% to ensure a high likelihood of order matching, even in volatile
 market situations. Your slippage is MEV protected.
 ```

**Review by josojo — APPROVED** [review]

File: src/custom/theme/index.tsx [hunk 3]
```diff
@@ -17,6 +22,30 @@ export enum Z_INDEX {
   tooltip = 1080,
 }
 
+type TextProps = TextPropsUni & { override?: boolean }
+
+export const TextWrapper = styled(Text)<{ color: keyof Colors; override?: boolean }>`
+  color: ${({ color, theme, override }) => {
+    const colour = (theme as any)[color]
```
Comment (alfetopito) [inline src/custom/theme/index.tsx:29]: `color` or `colour`?
Pick one :)

I know, I know, there's already a `color` const. Then `themeColor`?

**Review by alfetopito — APPROVED** [review]
Just a nitpick about colour because I'm a dick :D

**Review by elena-zh — COMMENTED** [review]
Hey @W3stside , the colored row looks good in the dark mode, but the yellow color in the text is not so bright and hard to read in the light mode. 
Could you please make it brighter/more readable, please? 
image

File: src/custom/theme/index.tsx (hunk 3 above)
Comment (W3stside) [reply src/custom/theme/index.tsx:29]: this is just a mod of theme i didn't code this, can we leave it as is pretty please?

File: src/cow-react/modules/swap/pure/Row/RowSlippageContent/index.tsx (hunk 2 above)
Comment (W3stside) [reply src/cow-react/modules/swap/pure/Row/RowSlippageContent/index.tsx:64]: will change in waterfall above this 👍

**Review by shoom3301 — APPROVED** [review]

File: src/custom/theme/index.tsx (hunk 3 above)
Comment (alfetopito) [reply src/custom/theme/index.tsx:29]: ok ok

**Comment by elena-zh** [comment]
> > Hey @W3stside , the colored row looks good in the dark mode, but the yellow color in the text is not so bright and hard to read in the light mode. Could you please make it brighter/more readable, please? image
> 
> > Hey @W3stside , the colored row looks good in the dark mode, but the yellow color in the text is not so bright and hard to read in the light mode. Could you please make it brighter/more readable, please? image
> 
> yeah I couldn't find a good one here, maybe @fairlighteth can help here?

Hi @W3stside , maybe this one?
image

**Comment by W3stside** [comment]
closing in favour of properly named #1239

**Pull request closed without merging**